.travis.yml

```diff
@@ -1,9 +1,32 @@
+sudo: required
+
 language: java
+
+jdk:
+  - oraclejdk8
+
 before_cache:
-- rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
-- rm -fr $HOME/.gradle/caches/*/plugin-resolution/
+  - rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
+  - rm -fr $HOME/.gradle/caches/*/plugin-resolution/
+
 cache:
   directories:
-  - $HOME/.gradle/caches/
-  - $HOME/.gradle/wrapper/
+    - $HOME/.gradle/caches/
+    - $HOME/.gradle/wrapper/
+
+branches:
+  only:
+    - master
+    - /^feature\/[a-zA-Z0-9\-]+$/
```

Comment:
Remove, there can also be bugfix and refactor branches that we want to build

Reply:
I vote for adding restrictions to ease up on travis resource overuse. If necessary the list can be modified whenever needed